src/benchmarks/micro/libraries/System.Collections/PriorityQueue/Perf_PriorityQueue.cs

```diff
@@ -0,0 +1,142 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Collections.Tests
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.Collections, Categories.GenericCollections)]
+    public abstract class Perf_PriorityQueue<TElement, TPriority>
```

Comment:
there is no need to introduce derived types, you can use the `GenericTypeArguments` attribute and BDN is going to instantiate the right thing for you.

```suggestion
    [GenericTypeArguments(typeof(int), typeof(int))]
    [GenericTypeArguments(typeof(string), typeof(string))]
    public class Perf_PriorityQueue<TElement, TPriority>
```

Reply:
Thanks, I wasn't aware of ValuesGenerator which is why I resorted to inheritance instead.